src/java/org/apache/cassandra/cql3/Operator.java

```diff
@@ -173,7 +387,24 @@ public String toString()
         public boolean isSatisfiedBy(AbstractType<?> type, ByteBuffer leftOperand, ByteBuffer rightOperand)
         {
             return type.compareForCQL(leftOperand, rightOperand) != 0;
+        }
+
+        @Override
+        public boolean requiresFilteringOrIndexingFor(ColumnMetadata.Kind columnKind)
+        {
+            return !columnKind.isPrimaryKeyKind();
```

Comment:
How would `!=` ever not require some kind of filtering?

Reply:
`b != 5` can use the clustering index

Reply:
I have “c != 5” there?